### DISCUSSION TITLE
DataWorkCloud 服务必须设置版本，请将属性 [wds.linkis.server.version] 添加到属性文件

### DISCUSSION THREAD

**Discussion by sunChuan** [WeBankFinTech/DataSphereStudio#1132] [General]
版本：1.1.2
部署exchangis时候报如下错误：
errCode: 10010 ,desc: DataWorkCloud service must set the version, please add property [wds.linkis.server.version] to properties file(DataWorkCloud 服务必须设置版本，请将属性 [wds.linkis.server.version] 添加到属性文件). ,ip: null ,port: 0 ,serviceKind: null

dss-exchagis-server.properties,已添加如下配置：
wds.linkis.gateway.ip=localhost
wds.linkis.gateway.port=9001
wds.linkis.gateway.url=http://localhost:9001/
wds.linkis.log.clear=true
wds.linkis.server.version=v1

请问哪一步出了问题

**Comment by MitchellChi** [comment]
1.config下的文件配置好后，将config下的所有文件复制到EXCHANGIS_CONF_PATH目录下。比如你在sbin/env.properties文件中配置EXCHANGIS_CONF_PATH="/appcom/config/exchangis-config/background"，则config下的所有文件都复制到/appcom/config/exchangis-config/background下,注意不要复制config目录，因为exchangis读取配置文件的位置是background

2.完成上面步骤再启动可能会遇到NoClassDefFoundError: org/springframework/jdbc/datasource/TransactionAwareDataSourceProxy，下载spring-jdbc.jar到lib/exchangis-server，spring-jdbc版本选择4.xx（4.3.8）即可

3.完成上面步骤再启动可能会遇到Failed to instantiate [com.netflix.discovery.EurekaClient]，是因为exchangis在引入spring-cloud-starter-netflix-eureka-client和spring-boot-starter-web两个依赖时出现冲突，将lib/exchangis-server下的jersey-client.jar和jersey-core.jar移走，注意lib/exchangis-server下下有两个版本的jersey-client.jar，都要移走。参考文章https://juejin.cn/post/7082351218940444686

此时重新启动服务应该正常

若还启动不了服务，控制台会打印出启动命令，将启动命令复制执行查看报错情况一步步排查，启动命令在
Launcher: launch to start server [xxx]下两行，类似
java/bin/java -Xms2g -Xmx2g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8 -XX:+UseG1GC -Xloggc:/opt/dss_linkis/appcom/logs/exchangis/background/dss-exchangis-main-server-dev-gc.log -XX:HeapDumpPath=/opt/dss_linkis/appcom/logs/exchangis/background/HeapDump/dss-exchangis-main-server-dev -Dspring.profiles.active=exchangis -DserviceName=dss-exchangis-main-server-dev -Dlog.path=/opt/dss_linkis/appcom/logs/exchangis/background -Dpid.file=/opt/dss_linkis/exchangis/wedatasphere-exchangis/sbin/../runtime/dss-exchangis-main-server-dev.pid -Dlogging.level.reactor.ipc.netty.channel.CloseableContextHandler=off -Duser.dir=/opt/dss_linkis/exchangis/wedatasphere-exchangis/sbin/../ -classpath /opt/dss_linkis/exchangis/wedatasphere-exchangis/sbin/../lib/exchangis-server/*:/opt/dss_linkis/appcom/config/exchangis-config/background:. com.webank.wedatasphere.exchangis.server.boot.ExchangisServerApplication
的一长串字符
